File: text_server.py
#!/usr/bin/python
#coding=utf8
from BaseHTTPServer import BaseHTTPRequestHandler,HTTPServer
import urllib
import cgi

class text_server(BaseHTTPRequestHandler):

    def do_GET(self):
        print "--------------------------------"
        print "Method: GET"
        print "Path: ", self.path
        print self.path
        path, param = urllib.splitquery(self.path)
        print 'path: ', path
        print 'param: ', param
        print "--------------------------------"

    def do_POST(self):
        print "--------------------------------"
        print "Method: POST"
        print "Path: ", self.path

        ctype, pdict = cgi.parse_header(self.headers.getheader('content-type'))
        print 'ctype: ', ctype
        print 'pdict: ', pdict
        if ctype == 'multipart/form-data':
            postvars = cgi.parse_multipart(self.rfile, pdict)
        elif ctype == 'application/x-www-form-urlencoded':
            length = int(self.headers.getheader('content-length'))
            postvars = cgi.parse_qs(self.rfile.read(length), keep_blank_values=1)
        else:
            postvars = {}

        if self.path == '/login':
            print 'pdata: ', postvars
            self.send_response(200)
            self.send_header('cookie', 'text_session')
            self.end_headers()
            self.wfile.write("{\"status\":1,\"user\": {\"id\": 666, \"authority\": 0, \"sex\": 0, \"driverType\": \"C1\", \"identify\": \"123\", \"phone\": \"phone\", \"photoURL\": \"photoURL\", \"address\": \"add\", \"companyID\": 4, \"appartmentID\": 78, \"jobNo\": 54}}");

        else:
            print self.path, " 意外的请求"
        print "--------------------------------"


PORT_NUMBER = 8888
import socket
myname = socket.getfqdn(socket.gethostname())
myaddr = socket.gethostbyname(myname)

try:
    server = HTTPServer((myaddr, PORT_NUMBER), text_server)
    print 'Started run at ', myaddr, PORT_NUMBER
    server.serve_forever()

except KeyboardInterrupt:
    print '^C received, shutting down the web server'
    server.socket.close()

File: text_server.js
function getIPAdress(){
    var interfaces = require('os').networkInterfaces();

    for(var devName in interfaces) {
        var iface = interfaces[devName];

        for(var i=0;i<iface.length;i++) {
            var alias = iface[i];

            if(alias.family === 'IPv4' && alias.address !== '127.0.0.1' && !alias.internal){
                return alias.address;
            }
        }
    }
}

http        = require('http');
url         = require('url');
querystring = require('querystring');

const hostname = getIPAdress();
const port = 8888;

const server = http.createServer((req, res) => {
    console.log(req.method)
    console.log(req.url)

    if (req.method == "POST" && req.url == "/login") {
        postData = '';
        req.on('data', function(data) {
            postData += data;
        });

        req.on('end', function() {
            console.log(querystring.parse(postData));
            res.statusCode = 200;
            res.setHeader('cookie', "sessionID");
            res.end("{\"status\":1,\"user\": {\"id\": 666, \"authority\": 0, \"sex\": 0, \"driverType\": \"C1\", \"identify\": \"123\", \"phone\": \"phone\", \"photoURL\": \"photoURL\", \"address\": \"add\", \"companyID\": 4, \"appartmentID\": 78, \"jobNo\": 54}}");
        });
    }

    else if (req.method == "GET") {
        res.statusCode = 200;
        var params = url.parse(req.url, true).query;
        console.log(params);
        res.statusCode = 200;

        if (req.url.split("?")[0] == "/getunit")
            res.end("{\"status\":1,\"unit\": {\"id\": 2, \"name\": \"UNIT NAME\"}}");
        else
            res.end("{\"status\":1,\"department\": {\"id\": 2, \"unitid\": 2, \"name\": \"UNIT NAME\"}}");
    }

});

server.listen(port, hostname, () => {
    console.log(`服务器运行在 http://${hostname}:8888/`);
});
